spec/wishlist.spec.js

```diff
@@ -0,0 +1,28 @@
+import Wishlist from '../src/api/wishlist';
+
+jest.mock('../src/api/base');
+
+describe('Wishlist should add a product to the wishlist', () => {
+    const mockedUrl = '/wishlist.php?action=add&wishlistid=14&product_id=111';
+
+    test('without options', () => {
+        const wishlist = new Wishlist();
+        const callback = jest.fn();
+
+        wishlist.itemAdd('14', '111', callback);
+
+        expect(wishlist.makeRequest).toHaveBeenCalledWith(mockedUrl, 'POST', {}, false, callback);
+    });
+
+    test('with options', () => {
+        const wishlist = new Wishlist();
+        const callback = jest.fn();
+        const mokedOptions = {
+            variantId: '10',
+        };
+
+        wishlist.itemAdd('14', '111', mokedOptions, callback);
+
+        expect(wishlist.makeRequest).toHaveBeenCalledWith(mockedUrl, 'POST', { variantId: '10' }, false, callback);
```

Comment:
Also you could check if the callback was called

Reply:
What do you thing on adding this check to request method which is responsible for calling callback? itemAdd responsible just for passing callback for further execution